### ISSUE TITLE
Multiline trailing comma rule flags type parameters

### ISSUE THREAD

**Issue by wmrowan** [palantir/tslint#1769]
### Bug Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.1.1`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```typescript
somefunction<
  TypeArg1,
  TypeArg2
>(
  arg1,
  arg2,
);
```

with `tslint.json` configuration:

```json
{
    "trailing-comma": [true, {"multiline": "always", "singleline": "never"}]
}
```

#### Actual behavior

TSLint flags line 3 in the sample code as a lint error. TSLint says that this line should have a trailing comma.

#### Expected behavior

TSLint should not flag this as a lint failure as a trailing comma is not allowed here by Typescript. Adding a comma to satisfy the linter will result in a Typescript compile error. While I think I personally agree with TSLint that a comma *should* be allowed here to be consistent with non-type function arguments, Typescript nevertheless disallows this syntax. While I think it would be reasonable to ask Typescript to fix this inconsistency, TSLint should probably be consistent with Typescript in the meantime.

**Comment by jmlopez-rod** [comment]
Perhaps more options need to be available, see [eslint comma-dangle](http://eslint.org/docs/rules/comma-dangle). With TSLint we need an option for generics.

**Comment by adidahiya** [comment]
yeah, makes sense, if it is not valid TypeScript then TSLint should not flag it as a violation. accepting PRs to fix this in the default rule behavior

**Comment by mizunashi-mana** [comment]
Missing such cases:

```typescript
var foo = <
    A,
    B
     ~ [Missing trailing comma]
>(a: A, b: B) => 42;
```

This issue should be reopened.
However, this may be caused by syntax parser.

**Comment by mizunashi-mana** [comment]
I found another case:

```typescript
function foo(a: [
  number,
  string
  ~ [Missing trailing comma] // expecting no warnings
]): void {
  return 42; 
}
```

**Comment by hitendramalviya** [comment]
Is there any workaround to get this fixed or we just need to  sicked with  `// tslint:disable-next-line:trailing-comma` ? But due to this my ts file looks ugly.

**Comment by adidahiya** [comment]
this was fixed by https://github.com/palantir/tslint/commit/a277873701ee24e75c8a742f83f054914ac20c65, available in v4.1+

**Comment by nchen63** [comment]
I tested @mizunashi-mana's 2 test cases above and they're not fixed in 4.3.1

**Comment by nchen63** [comment]
actually, #1905 was opened to cover those issues
